packages/mui-joy/src/Alert/AlertProps.ts

```diff
@@ -8,13 +9,44 @@ export interface AlertPropsVariantOverrides {}
 export interface AlertPropsColorOverrides {}
 export interface AlertPropsSizeOverrides {}
 
+interface ComponentsProps {
+  root?: SlotComponentProps<
+    'div',
+    { component?: React.ElementType; sx?: SxProps },
```

Comment:
The `component` and `sx` are present in all components' slots. It could be worth defining a Joy's version of `SlotComponentProps` with these props defined automatically.

Reply:
The `component` and `sx` are present in all components' slots. It could be worth defining a Joy's version of `SlotComponentProps` with these props defined automatically.

Reply:
That makes sense.

Reply:
I will work on this.

Reply:
Done in this [commit]([Create SlotComponentProps type for Joy UI and use it for typing of Jo…](https://github.com/mui/material-ui/pull/34997/commits/6148c4d56bcb7b1cd9c360fac687874a9b974011))